### DISCUSSION TITLE
Netflow ingestion - "Error template not found"

### DISCUSSION THREAD

**Discussion by AusAgentSmith** [akvorado/akvorado#1057] [Q&A]
Hi all,

I have hit a roadblock trying to get Akvorado to ingest opensense data, i can see the data is mkaing it to the container `"[docker-akvorado-inlet-1]"`

All i am seeing in the logs is,
`ERR netflow/root.go:198 > error while decoding NetFlow/IPFIX | error=NetFlowV9 NetFlowV9 [version:9 type:Decode obsDomainId:0: templateId:259] Error template not found`

I have trawled the doco and can't find any reference to if/how i am to specify a template.

any help would be greatly appreciated

**Comment by Sonicscream** [comment]
I'm having this same issue. In OPNsense I can set the netflow exporter to either V9 or V5 and I get the same error message as those flows are delivered. Akvorado seems to be the best solution available to use at the moment (that is free) and I would really love if this could get resolved.

Edit: I also have this issue with other netflow exporters. If there is a trick to resolve this, a nudge in the right direction would be appreciated.

**Reply by vincentbernat** [reply]
The troubleshooting documentation gives some hints to spot where the problem is.

**Comment by folkertvanheusden** [comment]
It says:

akvorado_inlet_flow_decoder_errors_total{name="netflow"} 3
akvorado_inlet_flow_decoder_netflow_errors_total{error="NetFlow/IPFIX
decoding error",exporter="172.29.0.11"} 3

So what should I do?

**Reply by vincentbernat** [reply]
Only 3 errors, so it is a transient errors and this is expected as said in the documentation. You should move forward in the documentation.

**Comment by folkertvanheusden** [comment]
It stays that way. Even after an hour: no "last flow" and "Flows/s" stays
0. Yet I see packets coming in via tcpdump.

**Reply by vincentbernat** [reply]
https://demo.akvorado.net/docs/troubleshooting#no-packets-exported

**Reply by folkertvanheusden** [reply]
Well the error seen (only 2 times: `akvorado_inlet_flow_decoder_netflow_errors_total{error="NetFlow/IPFIX decoding error",exporter="172.29.0.11"} 2`) is not listed.
Note that this is the only error for that ip-address and it doesn't increase. No output.

**Reply by vincentbernat** [reply]
That's not the only content of this documentation. Do you get anything from the first curl command?

**Reply by hslabbert** [reply]
> It stays that way. Even after an hour: no "last flow" and "Flows/s" stays 0. Yet I see packets coming in via tcpdump.

If you're still stuck:

opnsense does not appear to provide sample rate info in netflow v9. https://github.com/akvorado/akvorado/discussions/2000 is a good and detailed troubleshooting session, with the recommendation at https://github.com/akvorado/akvorado/discussions/2000#discussioncomment-14565834 also being the case for me of having to set a default sampling rate in order for enrichment to complete successfully for flows from opnsense, though I opted for `default-sampling-rate: 1` as the nf netflow FreeBSD package that opensense uses appears to be unsampled.
